node/pkg/processor/broadcast.go

```diff
@@ -50,20 +49,20 @@ func (p *Processor) broadcastSignature(
 	// Store our VAA in case we're going to submit it to Solana
 	hash := hex.EncodeToString(digest.Bytes())
 
-	if p.state.signatures[hash] == nil {
-		p.state.signatures[hash] = &state{
-			firstObserved: time.Now(),
-			nextRetry:     time.Now().Add(nextRetryDuration(0)),
-			signatures:    map[ethcommon.Address][]byte{},
-			source:        "loopback",
-		}
+	state, created := p.state.getOrCreateState(hash)
```

Comment:
how about using a different name than `state` here to not confuse it with `p.state`? Maybe `observationState` to signify that it's for one observation only or simply `s` to signify that it has a more local context.

Reply:
Renamed to `obsState`.